### PR TITLE
Remove references to deprecated application.yml config file.

### PR DESCRIPTION
**Pull request by achauve**
Remove references to deprecated application.yml config file
Point to .env and other .env.* config files instead.

#### What? Why?

- application.yml is not used any more but still present in docs and some files, and it misleads new developpers

#### What should we test?

- green build

#### Release notes
Remove references to deprecated application.yml config file

Changelog Category: Technical changes

#### Documentation updates

Following wiki pages need to be updated:

- https://github.com/openfoodfoundation/openfoodnetwork/wiki/Locale-and-sample-data -> mention correct .env file instead of application.yml
- https://github.com/openfoodfoundation/openfoodnetwork/wiki/Setting-up-Stripe-on-an-OFN-instance
- https://github.com/openfoodfoundation/openfoodnetwork/wiki/Set-and-Deploy-on-Heroku
- https://github.com/openfoodfoundation/openfoodnetwork/wiki/Development-Environment-Setup:-OS-X

### CODE REVIEW

**Comment by achauve** [comment]
To reviewers: is `config/schedule.rb` still used somewhere with `whenever`? or is another system is used instead (cron directly) for backups?
If `schedule.rb` is still used, we need to check that `application.yml` settings are also present in corresponding `.env` file.
After that, application.yml files should be deleted on instances (staging and production).

File: config/schedule.rb [hunk 1]
```diff
@@ -3,9 +3,7 @@
 
 # Learn more: http://github.com/javan/whenever
 
-app_config = YAML.load_file(File.join(__dir__, 'application.yml'))
```
Comment (mkllnk) [inline config/schedule.rb:6]: Whenever is still in use.

I just had a little test and we need to add some loading here or ENV isn't populated.

**Review by mkllnk — COMMENTED** [review]
Thank you!

Sorry, I ran out of time. I will look at this PR tomorrow again but there's something that needs changing.

**Comment by Matt-Yorkley** [comment]
> we need to check that `application.yml` settings are also present in corresponding `.env` file. 

Those variables get added to the .env file(s) via ofn-install in staging and production :+1:

> After that, application.yml files should be deleted on instances (staging and production).

We don't need to manually delete them from the servers, but we can maybe update ofn-install a bit to handle it. The issue there is that some instances that are not managed by the core team are behind on updates, so we currently still set the required variables in both files to maintain some backwards compatibility.

**Review by Matt-Yorkley — APPROVED** [review]
:+1:

**Comment by Matt-Yorkley** [comment]
Ah! There's a bit more we can do here;
- Remove `config/application.yml` from `.gitignore`
- Delete `config/application.yml`
- Delete `config/application.yml.example`

**Review by mkllnk — CHANGES_REQUESTED** [review]
Okay, looking at this again with a bit more time. This is great cleanup but I'll request some changes to complete this work:

* whenever needs fixing. I tested by adding something to my local env file and trying to access it via `puts` in config/schedule.rb and it didn't work. We need to load Dotenv correctly. I tried `require 'dotenv/rails-now'` but it only loaded `.env` and none of the other files. Can you try to figure this one out?
* Matt pointed out a few more places to clean up. I don't think that we can remove application.yml though because it's not tracked by Git.
* The testing steps should include removal from application.yml from staging, a deploy to staging where we check that the crontab is correct and maybe Google Maps works just to prove that we are accessing API keys correctly.

**Comment by achauve** [comment]
Thanks for the reviews!

@Matt-Yorkley Done what you suggested, but as Maikel noted, application.yml will be handled later by ansible as it's not versioned. 

@mkllnk Thanks for the indications on how whenever is used, I looked it up in ofn-install. And thanks for the test, indeed env was not loaded at all, and only '.env' when using `require 'dotenv/rails-now'`. Indeed I found that it's `dotenv-rails` which does all the work of finding all the .env* files but it requires all the rails environment normally loaded when you run `bundle exec rails some-cmd`. So adding a new require to load rails machinery seems fine. I tested it with different variables stored in .env, env.development and env.development.local .

As for the testing step on staging, how do you want to proceed @mkllnk @filipefurtad0? Is it a dev or a tester task?

File: config/schedule.rb [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# Force manual loading of rails application to get all env variables from dotenv-rails when running whenever cmd
+require File.expand_path('../environment',  __FILE__)
```
Comment (mkllnk) [inline config/schedule.rb:2]: I thought that we may need to do this. It's a shame to load so much code just for some config variables but at least this command is only used once per deploy.

**Review by mkllnk — APPROVED** [review]
Great!

If @filipefurtad0 is happy to log into the staging server and delete application.yml before deploying this PR then it can just be a normal test.

File: config/schedule.rb (hunk 2 above)
Comment (achauve) [reply config/schedule.rb:2]: Yes really bad in term of performance but as you say it's deploy time not execution time... the nice part is that in terms of code it's really lightweight.

**Comment by filipefurtad0** [comment]
Hey @achauve ,

Tested as suggested:
- deleted the `application.yml`
- I've logged into staging-UK

After deploying, performed some regression testing, by checking:
- configs were the same (locales, currencies, etc)
- maps were working as before - enterprises are correctly displayed and search function works as before

image

All good with these tests, so I'd say this is ready to go. However, I'm not entirely sure how to check if the check the crontab is correct. Any advice here?

Thank you!

**Comment by achauve** [comment]
Thanks @filipefurtad0 ! For the crontab I would say compare the output of running `crontab -l` on the machine as it was before, but it might be too late for that. Maybe compare the output on uk and on another staging instance which was deployed recently but with this branch?

File: config/schedule.rb (hunk 2 above)
Comment (Matt-Yorkley) [reply config/schedule.rb:2]: I wonder if it's even needed here, considering we effectively removed it months ago. At the same time we moved some other jobs into sidekiq scheduler.

If we're just scheduling some rake tasks, I think when those tasks actually run they will pick up the env vars as needed...? Know what I mean?

Comment (achauve) [reply config/schedule.rb:2]: I agree with you for `MAILTO` env var, but backups are scheduled only if  S3_BACKUPS_BUCKET is defined so this one seems mandatory.

**Comment by filipefurtad0** [comment]
Thank you for that advice @achauve ,
I'll repeat that test in another staging server and paste that output here :+1:

**Comment by filipefurtad0** [comment]
On staging-FR, before staging this PR:

```

openfoodnetwork@vps-2a25f087:~/apps/openfoodnetwork/current/config$ crontab -l


# Begin Whenever generated tasks for: ofn at: 2021-11-25 10:53:36 +0000
30 4 1 * * /bin/bash -l -c 'cd /home/openfoodnetwork/apps/openfoodnetwork/current && RAILS_ENV=staging bundle exec rake ofn:data:remove_transient_data --silent'

# End Whenever generated tasks for: ofn at: 2021-11-25 10:53:36 +0000

```

After deleting `application.yml` and deploying this PR:
```

openfoodnetwork@vps-2a25f087:~/apps/openfoodnetwork/current/config$ crontab -l


# Begin Whenever generated tasks for: ofn at: 2021-12-02 10:42:30 +0000
30 4 1 * * /bin/bash -l -c 'cd /home/openfoodnetwork/apps/openfoodnetwork/current && RAILS_ENV=staging bundle exec rake ofn:data:remove_transient_data --silent'

# End Whenever generated tasks for: ofn at: 2021-12-02 10:42:30 +0000
```

So, the same tasks are running **before and after** this PR.
Thanks for this @achauve and for updating the documentation! :tada: 

Merging.

File: config/schedule.rb (hunk 2 above)
Comment (Matt-Yorkley) [reply config/schedule.rb:2]: > backups are scheduled only if S3_BACKUPS_BUCKET is defined so this one seems mandatory.

I suspect in that case the rake task will just do nothing if there is no bucket, so the outcome is effectively the same (maybe with a tiny bit of inefficiency).

**Pull request merged**